DuckDuckGo/PrivacyProtectionController.swift

```diff
@@ -111,8 +111,9 @@ class PrivacyProtectionController: UIViewController {
 
     func updateSiteRating(_ siteRating: SiteRating?) {
         self.siteRating = siteRating
```

Comment:
What's the nullability contract of the SiteRating in the PrivacyProtectionController? It is currently unclear and this is the source of these kind of bugs. For example, this `updateSiteRating` method accepts an optional value and sets it to the `self.siteRating` however `self.siteRating` is force unwrapped which suggests it should never be null.

Imho the force unwrap here is bad (Apple's fault!) as we've lost all the compiler checks that would have caught this crash. If the siteRating can be optional then that's great we can change it and get the checks back but if it is meant to be non-optional we should see if we can massage this class to not force unwrap somehow.

Reply:
I'm not seeing the ambiguity - siteRating is optional, therefore can be nil (and it doesn't get force unwrapped).  

What am I missing?

Reply:
We discussed this and we're happy, but I'm going to raise a feedback item regarding the fact that the privacy grade screen might not reflect the underlying web view if it errors, etc.